src/constraints/GenericConstraint.jl

```diff
@@ -46,6 +50,17 @@ end
 
 vexity(c::GenericConstraint) = vexity(vexity(c.child), c.set)
 
+# The default vexity for MOI sets is assumed to required affine functions and be
+# convex. This is for the most common case of the connic sets. There might be
+# some incorrect answers, like if the user adds something like the `MOI.SOS1` or
```

Comment:
In that case, I think it's better to error than do something incorrect. I would simply have an error as default